File: backend/app/schemas/abbonamenti.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel


class CertificazioneCatalogoRead(BaseModel):
    id: UUID
    nome: str
    codice: str


class StatoCertificazioneRead(BaseModel):
    id: UUID
    nome: str


class AbbonamentoRead(BaseModel):
    certificazione_id: UUID
    certificazione_nome: str
    certificazione_codice: str
    stato_codice: str
    data_attivazione: date
    data_scadenza: date
    rinnovo_automatico: bool
    data_disattivazione: date | None
    updated_at: datetime


class AbbonamentoUpsert(BaseModel):
    stato_codice: str
    data_attivazione: date
    data_scadenza: date
    rinnovo_automatico: bool = True

File: frontend/lib/types/abbonamenti.ts
// Tipi TypeScript allineati ai contratti Pydantic esposti da
// backend/app/schemas/abbonamenti.py.

export interface CertificazioneCatalogo {
  id: string;
  nome: string;
  codice: string;
}

export interface StatoCertificazioneCatalogo {
  id: string;
  nome: string;
}

export interface Abbonamento {
  certificazione_id: string;
  certificazione_nome: string;
  certificazione_codice: string;
  stato_codice: string;
  data_attivazione: string;
  data_scadenza: string;
  rinnovo_automatico: boolean;
  data_disattivazione: string | null;
  updated_at: string;
}
